File: BRACU ACM Student Chapter/Online Workshop on Programming Skills DAY 03 ( C++ ) BRACU Data Structure/I.cpp
#include<bits/stdc++.h>
using namespace std;
priority_queue<int, vector<int>, greater<int>> store;

int main()
{
    int n;
    cin>>n;
    while (n!=0)
    {
        for(int i=0; i<n; i++)
        {
            int value;
            cin>>value;
            store.push(value);
        }

        int addagain=0;
        while(store.size()>1)
        {
            int a = store.top();    store.pop();
            int b = store.top();    store.pop();
            int sum = a+b;
            store.push(sum);
            addagain += sum;

        }
        cout << addagain <<endl;
        store.pop();
        cin>>n;
    }
    return 0;
}

File: BRACU ACM Student Chapter/Online Workshop on Programming Skills DAY 03 ( Java ) BRACU Data Structure/I.java

import java.util.PriorityQueue;
import java.util.Scanner;

public class I {

    static PriorityQueue<Integer> store;

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int test = input.nextInt();
        while (test != 0) {
            store = new PriorityQueue<>();
            for (int i = 0; i < test; i++) {
                store.add(input.nextInt());
            }

            int addagain = 0;
            while (store.size() > 1) {
                int a = store.poll();
                int b = store.poll();
                int sum = a + b;
                store.add(sum);
                addagain += sum;
            }
            System.out.println(addagain);
            test = input.nextInt();
        }
        input.close();
    }
}
